File: StringAlgorithm/T415 Add Strings/GO/main.go
package main

import (
	"fmt"
	"strconv"
)

func main() {
	num1 := "11"
	num2 := "123"
	fmt.Println("case1:", addStrings(num1, num2))

}

func addStrings(num1 string, num2 string) string {
	i := len(num1) - 1
	j := len(num2) - 1
	jinwei := 0
	ans := ""
	for i >= 0 || j >= 0 || jinwei != 0 {
		var x, y int
		if i >= 0 {
			x = int(num1[i] - '0')
		}
		if j >= 0 {
			y = int(num2[j] - '0')
		}

		res := x + y + jinwei
		ans = strconv.Itoa(res%10) + ans
		jinwei = res / 10
		i--
		j--
	}

	return ans
}

File: StringAlgorithm/T415 Add Strings/PYTHON/main.py
class Solution:
    def addStrings(self, num1: str, num2: str) -> str:
        num1_len = len(num1)
        num2_len = len(num2)
        jinwei = 0
        ans = ""
        for i in range(max(num1_len, num2_len)):
            if i < num1_len:
                num1_int = int(num1[num1_len - i - 1])
            else:
                num1_int = 0
            if i < num2_len:
                num2_int = int(num2[num2_len - i - 1])
            else:
                num2_int = 0
            sum = num1_int + num2_int + jinwei
            jinwei = sum // 10
            ans = str(sum % 10) + ans
        if jinwei == 1:
            ans = "1" + ans
        return ans